include/storage/serialization.hpp

```diff
@@ -266,14 +273,17 @@ template <typename K, typename V> void write(io::BufferWriter &writer, const std
     }
 }
 
-inline void read(io::BufferReader &reader, DataLayout &layout) { read(reader, layout.blocks); }
-
-inline void write(io::BufferWriter &writer, const DataLayout &layout)
+inline void read(io::BufferReader &reader, std::unique_ptr<BaseDataLayout> &layout)
```

Comment:
`unique_ptr&` makes no sense; the unique ptr carries ownership semantics. If all you want is a writeable ref to the object you can simply pass a `BaseDataLayout&` and the call site de-refs the unique ptr. Then it's clear on the call site and the function's semantics are clear, too.

It's a lot better in rust :P

https://doc.rust-lang.org/book/second-edition/ch04-02-references-and-borrowing.html

Reply:
Good idea - done.